goaviatrix/account.go

```diff
@@ -158,27 +157,26 @@ func (c *Client) ListAccounts() ([]Account, error) {
 	if err != nil {
 		return nil, err
 	}
-	accounts := resp.Results.AccountList
-
-	// Cache the result
-	c.cachedAccounts = accounts
+	c.cachedAccounts = resp.Results.AccountList
 
-	return accounts, nil
+	return c.cachedAccounts, nil
 }
 
-func (c *Client) GetAccount(account *Account) (*Account, error) {
+func (c *Client) GetAccount(account *Account) (Account, error) {
 	accList, err := c.ListAccounts()
 	if err != nil {
-		return nil, err
+		return Account{}, err
 	}
+	c.cacheMutex.Lock()
+	defer c.cacheMutex.Unlock()
 	for i := range accList {
 		if accList[i].AccountName == account.AccountName {
 			log.Infof("Found Aviatrix Account %s", account.AccountName)
-			return &accList[i], nil
+			return accList[i], nil
 		}
 	}
 	log.Errorf("Couldn't find Aviatrix account %s", account.AccountName)
-	return nil, ErrNotFound
+	return Account{}, ErrNotFound
```

Comment:
Nit: I prefer nil because of predictability, since default values change and easier to handle, but up you.